xblock/mixins.py

```diff
@@ -538,3 +540,51 @@ def index_dictionary(self):
         default implementation is an empty dict
         """
         return {}
+
+
+class ViewsMixin(object):
+    """
+    This mixin provides decorators that can be used on xBlock view methods.
+    """
+    @classmethod
+    def supports(cls, *functionalities):
+        """
+        A view decorator to indicate that an xBlock view has support for the
+        given functionalities.
+
+        Arguments:
+            functionalities: String identifiers for the functionalities of the view.
+                For example: "multi_device".
+        """
+        def _decorator(view):
+            """
+            Internal decorator that updates the given view's list of supported
+            functionalities.
+            """
+            # pylint: disable=protected-access
+            if not hasattr(view, "_supports"):
+                view._supports = set()
+            for functionality in functionalities:
+                view._supports.add(functionality)
+            return view
+        return _decorator
```

Comment:
There's nothing that enforces `functionalities` to be strings, which means integers, floats, lists, tuples, etc can all be passed in as functionalities. Is that the intent? Is it worth being concerned about?